.github/workflows/pull_request.yaml

```diff
@@ -2,57 +2,84 @@ name: Pull Request
 on:
   pull_request:
     types: [opened, reopened, synchronize]
+
 permissions:
   contents: read
   pull-requests: write
+
 env:
   BASE_ENVIRONMENT: 'development'
   ENVIRONMENT_TYPE: 'development'
-  HUMCTL_VERSION: '0.23.2'
   ENVIRONMENT_ID: pr-${{ github.event.number }}
   ENVIRONMENT_NAME: PR-${{ github.event.number }}
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+      
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GH_TOKEN }}
+
       - uses: humanitec/setup-cli-action@v1
         with:
-          version: ${{ env.HUMCTL_VERSION }}
+          version: ${{ vars.HUMCTL_VERSION }}
+
       - name: Create Humanitec Env
+        if: ${{ github.event.action == 'opened' || github.event.action == 'reopened' }}
         run: |
-            humctl create environment ${{ env.ENVIRONMENT_ID }} \
-              --token ${{ secrets.HUMANITEC_TOKEN }} \
-              --org ${{ vars.HUMANITEC_ORG }} \
-              --app ${{ vars.APP_NAME }} \
-              --name ${{ env.ENVIRONMENT_NAME }} \
-              -t ${{ env.ENVIRONMENT_TYPE }} \
-              --from ${{ env.BASE_ENVIRONMENT }} \
-              || true
-      - uses: score-spec/setup-score@v2
-        with:
-          file: score-humanitec
-          version: ${{ env.SCORE_HUMANITEC_VERSION }}
+          humctl create environment ${{ env.ENVIRONMENT_ID }} \
+            --token ${{ secrets.HUMANITEC_TOKEN }} \
+            --org ${{ vars.HUMANITEC_ORG }} \
+            --app ${{ vars.APP_NAME }} \
+            --name ${{ env.ENVIRONMENT_NAME }} \
+            -t ${{ env.ENVIRONMENT_TYPE }} \
+            --from ${{ env.BASE_ENVIRONMENT }}
+        
+      - name: Build and push image
+        id: build_push
+        run: |
+          # Make sure repo name is all lowercase for docker push
+          GH_REPOSITORY=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
+
+          TAG=$GH_REPOSITORY:${{ env.ENVIRONMENT_ID }}
```

Comment:
Depending on the `imagePullPolicy` in the cluster, if multiple images are built via this PR, the updates in this image may not be reflected in the running container... not a big deal as we are mostly illustrating Score/Humanitec... but just for your info.
Another way could be to still keep this tag, and in Score passing the `digest`, something around this:
```
oras manifest fetch ${{ env.IMAGE_NAME }}:${{ env.IMAGE_TAG }} \
              --descriptor \
              | jq -r .digest
```

Reply:
Yes, good point. I'll check if I can obtain the image SHA256 work and then replace `<image-name>:<tag>` with `<image-name>@<digest>` in the `score deploy`.

Reply:
See the change below (line 62): we now get the digest via a `docker inspect` (preventing the `oras` dependency) and pass it on to the Score deploy step.